ipatests/test_integration/test_user_permissions.py

```diff
@@ -93,3 +96,57 @@ def test_stageuser_show_as_alternate_admin(self):
         # the field Kerberos Keys available must contain True
         result = self.master.run_command(['ipa', 'stageuser-show', stageuser])
         assert 'Kerberos keys available: True' in result.stdout_text
+
+
+class TestInstallClientNoAdmin(IntegrationTest):
+    num_clients = 1
+
+    def test_installclient_as_user_admin(self):
+        """ipa-client-install should not use hardcoded admin for principal
+        Related to : https://pagure.io/freeipa/issue/5406
+        """
+        client = self.clients[0]
+        tasks.install_master(self.master)
+        tasks.kinit_admin(self.master)
+        username = 'testuser1'
+        password = 'userSecretPassword123'
+        password_confirmation = "%s\n%s\n" % (password,
+                                              password)
+
+        self.master.run_command(['ipa', 'user-add', username,
+                                 '--first', username,
+                                 '--last', username,
+                                 '--password'],
+                                stdin_text=password_confirmation)
+
+        runcmd1 = ['ipa', 'role-add', 'useradmin']
+        self.master.run_command(runcmd1)
+        stdin_text = "useradmin\nHost Enrollment\n"
+        self.master.run_command(['ipa', 'role-add-privilege'], stdin_text)
```

Comment:
Adding the privilege will fail. It neds to look more like (untested):

stdin_text = "useradmin\n"
 self.master.run_command(['ipa', 'role-add-privilege', '--privilege=Host Enrollment'], stdin_text)